cmd/gateway/s3/gateway-s3.go

```diff
@@ -485,6 +487,30 @@ func (l *s3Objects) CopyObject(ctx context.Context, srcBucket string, srcObject
 	for k, v := range header {
 		srcInfo.UserDefined[k] = v[0]
 	}
+	tags, ok := srcInfo.UserDefined[xhttp.AmzObjectTagging]
+	if ok {
+		delete(srcInfo.UserDefined, xhttp.AmzTagDirective)
+		delete(srcInfo.UserDefined, xhttp.AmzObjectTagging)
+		delete(srcInfo.UserDefined, xhttp.AmzTagCount)
+		delete(srcInfo.UserDefined, strings.ToLower(xhttp.AmzMetadataDirective))
+		delete(srcInfo.UserDefined, strings.ToLower(xhttp.AmzCopySourceIfMatch))
+		var copyDstInfo miniogo.DestinationInfo
+		var err error
+		var tag tagging.Tagging
+		if tag, err = tagging.FromString(tags); err != nil {
+			return minio.ObjectInfo{}, err
+		}
+		copySrcInfo := miniogo.NewSourceInfo(srcBucket, srcObject, srcOpts.ServerSideEncryption)
+		if copyDstInfo, err = miniogo.NewDestinationInfoWithOptions(dstBucket, dstObject, miniogo.DestInfoOptions{
+			UserTags: tagging.ToMap(tag), ReplaceTags: true, Encryption: dstOpts.ServerSideEncryption, UserMeta: srcInfo.UserDefined,
+		}); err != nil {
```

Comment:
You can create a new var for `miniogo.DestInfoOptions` and pass it to NewDestinationInfoWithOptions, that way it will be more readable